torchdynamo/symbolic_convert.py

```diff
@@ -1279,16 +1305,18 @@ def inline_call_(parent, func, args, kwargs):
 
         if is_generator(code):
             tracer = InliningGeneratorInstructionTranslator(
-                parent, code, sub_locals, closure_cells, func
+                parent, code, sub_locals, parent.symbolic_globals, closure_cells, func
             )
         else:
             tracer = InliningInstructionTranslator(
-                parent, code, sub_locals, closure_cells, func
+                parent, code, sub_locals, parent.symbolic_globals, closure_cells, func
             )
 
         tracer.run()
         assert tracer.symbolic_result is not None
         func.export_freevars(parent, tracer)
+        # Merge symbolic_globals to the parent
+        parent.symbolic_globals.update(tracer.symbolic_globals)
```

Comment:
We should only do this is we are in the same namespace as the parent.  `f_globals is parent.f_global`

If these come from different files they will have different globals.